src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorage/AzureTableStorageSink.cs

```diff
@@ -36,9 +38,30 @@ public class AzureTableStorageSink : ILogEventSink
         /// <param name="storageAccount">The Cloud Storage Account to use to insert the log entries to.</param>
         /// <param name="formatProvider">Supplies culture-specific formatting information, or null.</param>
         /// <param name="storageTableName">Table name that log entries will be written to. Note: Optional, setting this may impact performance</param>
-        public AzureTableStorageSink(CloudStorageAccount storageAccount, IFormatProvider formatProvider, string storageTableName = null)
+        public AzureTableStorageSink(
+            CloudStorageAccount storageAccount,
+            IFormatProvider formatProvider,
+            string storageTableName
+            ) :this(storageAccount, formatProvider, storageTableName, DefaultRowKeyFormatter)
+        {
+        }
+
+        /// <summary>
+        /// Construct a sink that saves logs to the specified storage account.
+        /// </summary>
+        /// <param name="storageAccount">The Cloud Storage Account to use to insert the log entries to.</param>
+        /// <param name="formatProvider">Supplies culture-specific formatting information, or null.</param>
+        /// <param name="storageTableName">Table name that log entries will be written to. Note: Optional, setting this may impact performance</param>
+        /// <param name="rowKeyFormatter">formater for the rowkey, func will be paseed th current rowkey and a unique number and should return
+        /// the value of the row key to store</param>
+        public AzureTableStorageSink(
+            CloudStorageAccount storageAccount,
+            IFormatProvider formatProvider,
+            string storageTableName = null,
+            Func<string, long, string> rowKeyFormatter = null)
```

Comment:
I think it's going to be easier to consume this if it's an interface rather than a func because otherwise it's not immediately clear what each of the parameters are.